### PR TITLE
[RFC] [tryfix] segmented ttml

### PR DESCRIPTION
**Pull request by stefansaraev**
while working on an addon for local VOD provider, I noticed subtitles were not working at all.

they are referenced in an ism manifest as
```
<StreamIndex Type="text" Language="BUL" Name="textstream_BUL_CAPT" Subtype="CAPT" Chunks="61" QualityLevels="1" Url="QualityLevels({bitrate})/Fragments(textstream_BUL_CAPT={start time})">
```

fragment sample: http://sprunge.us/TeDJ

this is a quick (incomplete but works for me (tm)) attempt to adapt the TTML parser.

### CODE REVIEW

**Comment by stefansaraev** [comment]
removed an early reurn in TTML2SRT::Parse which caused subtitles to be rendered all over the screen.

File: src/parser/TTML.cpp [hunk 1]
```diff
@@ -208,12 +208,9 @@ bool TTML2SRT::Parse(const void *buffer, size_t buffer_size, uint64_t timescale,
   XML_SetElementHandler(parser, start, end);
   XML_SetCharacterDataHandler(parser, text);
 
-  XML_Status retval = XML_Parse(parser, (const char*)buffer, buffer_size, done);
+  XML_Parse(parser, (const char*)buffer, buffer_size, done);
   XML_ParserFree(parser);
 
-  if (retval == XML_STATUS_ERROR)
```
Comment (peak3d) [inline src/parser/TTML.cpp]: I don't understand this removal

Comment (stefansaraev) [reply src/parser/TTML.cpp]: without this removal, (non-english) subtitles are rendered once they are parsed like this: http://imgur.com/a/wFkcO

I could not find the root cause, but removing the return fixes the issue

here is a [bulgarian](https://test.net.in/Fragments(textstream_BUL_CAPT=14400000000)) fragment sample (does not match the screenshot above but you get the idea) that broke

another [english](https://test.net.in/Fragments(textstream_eng_SUBT=14400000000)) sample that worked

Comment (stefansaraev) [reply src/parser/TTML.cpp]: just to add, as a workaround, we adapted our addon to download external .dfxp subtitle file, convert it to srt and and tell the player to use load external subs. so this is not high prio. would be nice if you have an idea what's the root cause :)

Comment (peak3d) [reply src/parser/TTML.cpp]: ok, thanks for providing samples!

Comment (stefansaraev) [reply src/parser/TTML.cpp]: first commit is still needed. it makes (at least) english subs work. I can drop the second one for now if you want me to.

**Comment by peak3d** [comment]
let me read first a little bit about expat / namespaces

**Comment by stefansaraev** [comment]
@peak3d updated. moved the return where it belongs. should be ok now ?

**Comment by stefansaraev** [comment]
better fix was pushed to master, I'll close this. thanks @peak3d

**Pull request closed without merging**